crates/uv/tests/tool_install.rs

```diff
@@ -319,6 +319,34 @@ fn tool_install_version() {
     "###);
 }
 
+/// Test an editable install of a tool
+#[test]
+fn tool_install_editable() {
+    let context = TestContext::new("3.12").with_filtered_exe_suffix();
+    let tool_dir = context.temp_dir.child("tools");
+    let bin_dir = context.temp_dir.child("bin");
+
+    // Install `black`
+    uv_snapshot!(context.filters(), context.tool_install()
+        .arg("-e")
+        .arg(context.workspace_root.join("scripts/packages/black_editable"))
+        .env("UV_TOOL_DIR", tool_dir.as_os_str())
+        .env("XDG_BIN_HOME", bin_dir.as_os_str())
+        .env("PATH", bin_dir.as_os_str()), @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    No executables are provided by `black`
+
+    ----- stderr -----
+    warning: `uv tool install` is experimental and may change without warning
+    Resolved 1 package in [TIME]
+    Prepared 1 package in [TIME]
+    Installed 1 package in [TIME]
+     + black==0.1.0 (from file://[WORKSPACE]/scripts/packages/black_editable)
+    "###);
```

Comment:
Can we also add a test for `uv tool install back` after `uv tool install -e ./scripts/packages/black_editable`?

Reply:
> Can we show the receipt here?

`uv tool install` exits with a failure for the `black_editable` package since it has no executables. So no receipt gets created in this scenario.

Reply:
Can we add an executable to it?

Reply:
I've added one a simple executable and added further additional snapshot tests.

> Can we also add a test for uv tool install back after uv tool install -e ./scripts/packages/black_editable?

I added this too, but it just installs `black` version from the PyPi in this case, do you think that's the right desired here?

Reply:
I think that's because you're creating a new context here. If you remove that, it retains the install (but the behavior is kind of confusing -- https://github.com/astral-sh/uv/pull/5489 helps).